### PR TITLE
Deploy fritzbox

### PR DESCRIPTION
**Pull request by fritteli**
This is a deploy hook for the FRITZ!Box routers produced by AVM (https://avm.de/produkte/fritzbox/).

### CODE REVIEW

File: deploy/fritzbox.sh [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  printf -- "--${_post_boundary}\r\n" >> "${_post_request}"
+  printf "Content-Disposition: form-data; name=\"BoxCertImportFile\"; filename=\"BoxCert.pem\"\r\n" >> "${_post_request}"
+  printf "Content-Type: application/octet-stream\r\n\r\n" >> "${_post_request}"
+  cat "${_ckey}" >> "${_post_request}"
+  cat "${_cfullchain}" >> "${_post_request}"
+  printf "\r\n" >> "${_post_request}"
+  printf -- "--${_post_boundary}--" >> "${_post_request}"
+
+  _info "Upload certificate to the FRITZ!Box"
+  wget --no-check-certificate -q -O - "${_fritzbox_url}/cgi-bin/firmwarecfg" --header="Content-type: multipart/form-data boundary=${_post_boundary}" --post-file "${_post_request}" | grep SSL
```
Comment (Neilpang) [inline deploy/fritzbox.sh]: do not use  `wget` or `curl` directly.
please use the `_get()` or `_post()` function instead.

if you want `--no-check-certificate` option,  please define `HTTPS_INSECURE=1`

```sh
export HTTPS_INSECURE=1
_get  https:///...........
```

**Comment by Neilpang** [comment]
please fix the formatting errors first https://travis-ci.org/Neilpang/acme.sh/jobs/269794010

**Comment by fritteli** [comment]
The build has now finally passed. The pull request should be ready to be merged now. Thanks for your patience!

**Comment by fritteli** [comment]
I've resolved the merge conflicts.

File: deploy/fritzbox.sh (hunk 1 above)
Comment (fritteli) [reply deploy/fritzbox.sh]: Done.

**Comment by hebbet** [comment]
works like a charm

**Pull request merged**